var/spack/repos/builtin/packages/kokkos-kernels/package.py

```diff
@@ -208,7 +208,7 @@ class KokkosKernels(CMakePackage, CudaPackage):
     for eti in space_etis:
         deflt, descr, backend_required = space_etis[eti]
         variant(eti, default=deflt, description=descr)
-        depends_on("kokkos+%s" % backend_required, when="+%s" % eti)
+        depends_on(f"kokkos+{backend_required}", when=f"{eti}=auto")
```

Comment:
@lucbv should this variant actually be a boolean? Can it ever be anything other than auto?

Reply:
Yeah, I am not quite sure what the intent was here with the first field in {exe,mem}space. At first glance I would say it is always auto?

Reply:
This fixes the current implementation that currently only uses `auto` as far as Spack is concerned. I think the maintainers can refactor it later to either be a bool or add more values in the variant.